### PR TITLE
New "Also share on Audio.com dialog" (#19487)

### PR DESCRIPTION
**Pull request by mathesoncalum**
Resolves: #19487

Also includes some method/variable renaming for clarity.

### CODE REVIEW

File: src/cloud/audiocom/audiocomservice.cpp [hunk 1]
```diff
@@ -372,6 +372,11 @@ Ret AudioComService::doCreateAudio(network::INetworkManagerPtr manager, const QS
     return ret;
 }
 
+QUrl AudioComService::audioComUrl() const
```
Comment (Eism) [inline src/cloud/audiocom/audiocomservice.cpp:375]: You can get this info through AudioComService::cloudInfo().url, so don't need to create a new method

File: src/project/qml/MuseScore/Project/AlsoShareAudioComDialog.qml [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+            id: header
+
+            width: parent.width
+            spacing: 16
+
+            StyledTextLabel {
+                Layout.fillWidth: true
+
+                //: The text between `<a href=\"%1\">` and `</a>` will be a clickable link to Audio.com
+                text: qsTrc("project/cloud", "Would you also like to share your music on <a href=\"%1\">Audio.com</a>?").arg(root.audioComUrl)
```
Comment (Eism) [inline src/project/qml/MuseScore/Project/AlsoShareAudioComDialog.qml:66]: We need to voice the entire dialog text. See

https://github.com/musescore/MuseScore/blob/5f57ce04e0978db91f6195a2ea6f92a4030a4b5b/src/framework/ui/qml/MuseScore/Ui/StandardDialogPanel.qml#L126

Also 

https://github.com/musescore/MuseScore/blob/5f57ce04e0978db91f6195a2ea6f92a4030a4b5b/src/framework/ui/qml/MuseScore/Ui/StandardDialogPanel.qml#L225

File: src/project/internal/projectactionscontroller.cpp [hunk 3]
```diff
@@ -962,6 +967,36 @@ bool ProjectActionsController::saveProjectToCloud(CloudProjectInfo info, SaveMod
...
 }
 
+void ProjectActionsController::alsoShareAudioCom(const AudioFile& audio)
+{
+    if (!configuration()->showAlsoShareAudioComDialog()) {
+        shareAudio(audio);
+        return;
+    }
+
+    QUrl audioComUrl = audioComService()->cloudInfo().url;
```
Comment (Eism) [inline src/project/internal/projectactionscontroller.cpp:977]: You can use cloudInfo from authorization
`audioComService()->authorization()->cloudInfo().url;`

File: src/project/qml/MuseScore/Project/AlsoShareAudioComDialog.qml [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+            }
+        }
+    }
+
+    AccessibleItem {
+        id: accessibleInfo
+
+        accessibleParent: navPanel.accessible
+        role: MUAccessible.Button
+        name: titleInfo.text + "; " + subtitleInfo.text + "; " + preferenceInfo.text
```
Comment (Eism) [inline src/project/qml/MuseScore/Project/AlsoShareAudioComDialog.qml:88]: Цe shouldn't include preferenceInfo in the text. 
This text will be heard by the user when they navigate to the checkbox, while the other text will not be navigable for the user. That's why we're adding AccessibleInfo for these texts.

File: src/project/qml/MuseScore/Project/AlsoShareAudioComDialog.qml [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+
+        RowLayout {
+            id: options
+
+            CheckBox {
+                id: checkbox
+
+                Layout.fillWidth: true
+
+                navigation.panel: navPanel
```
Comment (Eism) [inline src/project/qml/MuseScore/Project/AlsoShareAudioComDialog.qml:159]: I think it should be in a separate panel than the ButtonBox and the user will be able to navigate via Tab.
Also please set `navigation.row`, at the moment this checkbox is not available for navigation as I see

**Comment by zacjansheski** [comment]
Tested on Mac OS 14, Ubuntu 20.04.6, Windows 11.  Approved.

https://github.com/musescore/MuseScore/issues/19487 FIXED

**Pull request merged**